### PR TITLE
added use of useQueryParameters in BirthdayReportPage

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/celebrations/Birthdays.jsx [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import React, { useContext } from 'react';
 import { Link } from 'react-router-dom';
 
+import { Card, CardHeader, Grid, Typography } from '@mui/material';
```
Comment (mvolkmann) [inline web-ui/src/components/celebrations/Birthdays.jsx:4]: The only changes in this file were to reorder/combine some imports. I thought I was going to need other changes in this file, but it turned out none were required.

File: web-ui/src/pages/BirthdayReportPage.jsx [hunk 2]
```diff
@@ -13,33 +13,47 @@ import SearchBirthdayAnniversaryResults from '../components/search-results/Searc
 import { sortBirthdays } from '../context/util';
 
 import { selectCsrfToken } from '../context/selectors';
+import { useQueryParameters } from '../helpers/query-parameters';
 
 const months = [
-  { month: 'January' },
```
Comment (mvolkmann) [inline web-ui/src/pages/BirthdayReportPage.jsx:18]: Using objects with a `month` property was making this code more complicated than it needed to be. This is now just an array of strings.

File: web-ui/src/pages/BirthdayReportPage.jsx [hunk 3]
```diff
@@ -13,33 +13,47 @@ import SearchBirthdayAnniversaryResults from '../components/search-results/Searc
...
+      value: selectedMonths,
+      setter: setSelectedMonths,
+      toQP(value) {
+        return value ? value.join(',') : [];
+      }
+    }
+  ]);
+
   const handleSearch = async monthsToSearch => {
-    const months = monthsToSearch.map(m => m.month);
```
Comment (mvolkmann) [inline web-ui/src/pages/BirthdayReportPage.jsx:41]: We no longer need to get the `month` property out of an array of objects. We already have an array of strings.

File: web-ui/src/pages/BirthdayReportPage.jsx [hunk 4]
```diff
@@ -55,13 +69,12 @@ const BirthdayReportPage = () => {
...
           options={months}
-          defaultValue={[months[currentMonth].month]}
+          defaultValue={defaultMonths}
           value={selectedMonths}
           onChange={onMonthChange}
           isOptionEqualToValue={(option, value) => {
-            return value ? value.month === option.month : false;
+            return value ? value === option : false;
           }}
-          getOptionLabel={option => option.month}
```
Comment (mvolkmann) [inline web-ui/src/pages/BirthdayReportPage.jsx:64]: We no longer need the `getOptionLabel` property since the options are already the strings we want.

**Review by mkimberlin — APPROVED** [review]

**Review by vhscom — APPROVED** [review]

**Pull request merged**